packages/dynamodb/src/index.ts

```diff
@@ -1,447 +1,426 @@
-            GSI1PK: `ACCOUNT#${providerAccountId}`,
-            providerId,
-            providerAccountId,
-            providerType,
-            refreshToken,
-            accessToken,
-            accessTokenExpires,
-            type: "ACCOUNT",
-            userId,
-            createdAt: now.toISOString(),
-            updatedAt: now.toISOString(),
-          }
-
-          await client.put({ TableName, Item: item }).promise()
-          return item as any
+      })
+      const user = res.Item
+      return user ? format.from<AdapterUser>(user) : null
+    },
+    async updateUser(user) {
+      const {
+        UpdateExpression,
+        ExpressionAttributeNames,
+        ExpressionAttributeValues,
+      } = generateUpdateExpression(user)
+      const data = await client.update({
+        TableName,
+        Key: {
+          // next-auth type is incorrect it should be Partial<AdapterUser> & {id: string} instead of just Partial<AdapterUser>
+          pk: `USER#${user.id as string}`,
+          sk: `USER#${user.id as string}`,
         },
-        async unlinkAccount(userId, providerId, providerAccountId) {
-          const deleted = await client
-            .delete({
-              TableName,
-              Key: {
-                pk: `USER#${userId}`,
-                sk: `ACCOUNT#${providerId}#${providerAccountId}`,
-              },
-            })
-            .promise()
-
-          return deleted
+        UpdateExpression,
+        ExpressionAttributeNames,
+        ExpressionAttributeValues,
+        ReturnValues: "ALL_NEW",
+      })
+      const attributesDynamo = data.Attributes as Dynamo<AdapterUser>
+      return format.from<AdapterUser>(attributesDynamo)
+    },
+    async deleteUser(userId) {
+      // query all the items related to the user to delete
+      const resQuery = await client.query({
+        TableName,
+        KeyConditionExpression: "#pk = :pk",
+        ExpressionAttributeNames: { "#pk": "pk" },
+        ExpressionAttributeValues: { ":pk": `USER#${userId}` },
+      })
+      if (!resQuery.Items) {
+        return null
+      }
+      const items = resQuery.Items
+      // find the user we want to delete to return at the end of the function call
+      const user = items.find((item) => item.type === "USER")
+      const itemsToDelete = items.map((item) => {
+        return {
+          DeleteRequest: {
+            Key: {
+              sk: item.sk,
+              pk: item.pk,
+            },
+          },
+        }
+      })
+      // batch write commands cannot handle more than 25 requests at once
+      const itemsToDeleteMax = itemsToDelete.slice(0, 25)
+      const param: BatchWriteCommandInput = {
+        RequestItems: { [TableName]: itemsToDeleteMax },
+      }
+      await client.batchWrite(param)
+      return user ? format.from<AdapterUser>(user) : null
+    },
+    async linkAccount(data) {
+      const item = {
+        ...data,
+        id: randomBytes(16).toString("hex"),
```

Comment:
Can you please help me understand the purpose of this `id` field? It doesn't appear to be doing anything.

Reply:
It doesn't do much for sure. It's mainly there to reference the id of the user directly. I've seen it's considered best practice in Single table design to store all attributes outside of the sk, pk, gsk1, gpk1 etc... The reason is that those attributes are composite keys that are created from other fields and it can get pretty messy to understand when those composite keys are created. So here we could find out the id of the user from the field pk by removing USER# from that field but if you change the access patterns pk might be something more complicated

Reply:
I think It's mostly there to make typescript happy 😂

Reply:
Oh no 😂😭 Do you think it's the same thing on `createSession`?

Reply:
I just looked at the types and we do have
```javascript
export interface AdapterSession {
  id: string
....
```
so even if that id is not really used anywhere I think it's better to add it to be safe in case it's used at some point. 

But looking at the Account interface there doesn't seem to be any id there. It might be safe to remove it? I think I might have left it because it was there in the previous version of the dynamo adapter...